### PR TITLE
Account, Role, Permission entities w/ repos, services, controllers + Security + Swagger

### PR DESCRIPTION
**Pull request by TarasOliinyk**

### CODE REVIEW

File: src/main/java/com/lits/tovisitapp/controller/RoleController.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    @HasUpdatePermission
+    @PutMapping("/{id}/addPermissions")
+    public ResponseEntity<RoleDTO> addPermissionsToRole(@PathVariable (name = "id") Long roleId,
+                                                        @RequestBody List<Long> permissionIds) {
+        RoleDTO roleDTO = roleService.getRoleById(roleId);
+        List<PermissionDTO> permissionDTOs = permissionIds.stream().map(permissionService::getPermissionById)
+                .collect(Collectors.toList());
+                roleDTO.getPermissions().addAll(permissionDTOs);
+        return ResponseEntity.status(HttpStatus.OK).body(roleService.updateRole(roleDTO));
```
Comment (natalia-kozak) [inline src/main/java/com/lits/tovisitapp/controller/RoleController.java]: Do this in the Service layer

File: src/main/java/com/lits/tovisitapp/controller/RoleController.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    @HasUpdatePermission
+    @PutMapping("/{id}/removePermissions")
+    public ResponseEntity<RoleDTO> removePermissionsFromRole(@PathVariable (name = "id") Long roleId,
+                                                             @RequestBody List<Long> permissionIds) {
+        RoleDTO roleDTO = roleService.getRoleById(roleId);
+        List<PermissionDTO> permissionDTOs = permissionIds.stream().map(permissionService::getPermissionById)
+                .collect(Collectors.toList());
+        roleDTO.getPermissions().removeAll(permissionDTOs);
+        return ResponseEntity.status(HttpStatus.OK).body(roleService.updateRole(roleDTO));
+    }
```
Comment (natalia-kozak) [inline src/main/java/com/lits/tovisitapp/controller/RoleController.java]: Do this in the Service layer

File: src/main/java/com/lits/tovisitapp/controller/AccountController.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    @HasUpdatePermission
+    @PutMapping("/{accountId}/removeRole/{roleId}")
+    public ResponseEntity<AccountDTO> unassignRole(@PathVariable (name = "accountId") Long accountId,
+                                                   @PathVariable (name = "roleId") Long roleId) {
+        RoleDTO roleToRemove = roleService.getRoleById(roleId);
+        AccountDTO accountDTO = accountService.getAccountById(accountId);
+        accountDTO.getRoles().remove(roleToRemove);
+        return ResponseEntity.status(HttpStatus.OK).body(accountService.updateAccount(accountDTO));
+    }
```
Comment (natalia-kozak) [inline src/main/java/com/lits/tovisitapp/controller/AccountController.java]: Do this in the Service layer

File: src/main/java/com/lits/tovisitapp/controller/AccountController.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    @HasUpdatePermission
+    @PutMapping("/{accountId}/addRoles")
+    public ResponseEntity<AccountDTO> assignRoles(@PathVariable (name = "accountId") Long accountId,
+                                                  @RequestBody List<Long> roleIds) {
+        AccountDTO accountDTO = accountService.getAccountById(accountId);
+        List<RoleDTO> roleDTOs = roleService.getRolesWithIds(roleIds);
+        accountDTO.getRoles().addAll(roleDTOs);
+        return ResponseEntity.status(HttpStatus.OK).body(accountService.updateAccount(accountDTO));
+    }
```
Comment (natalia-kozak) [inline src/main/java/com/lits/tovisitapp/controller/AccountController.java]: Do this in the Service layer

**Review by natalia-kozak — APPROVED** [review]

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -2,7 +2,7 @@ spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 spring.datasource.username=root
 spring.datasource.password=Mypass@123
 spring.datasource.url=jdbc:mysql://localhost:3306/tovisitappdb?createDatabaseIfNotExist=true&serverTimezone=UTC
-spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.hibernate.ddl-auto=update
```
Comment (Volodymyr95) [inline src/main/resources/application.properties:5]: delete

**Pull request closed without merging**